File: frameworks/runtime-src/proj.android-studio/app/src/com/gggame/quanguonn/utils/TalkingDataUtil.java
package com.gggame.quanguonn.utils;

import com.tendcloud.tenddata.TDGAAccount;
import com.tendcloud.tenddata.TDGAVirtualCurrency;

/**
 * talkingdata工具
 * Created by zhangluxin on 2017/11/1.
 */
public class TalkingDataUtil {

    /**
     * 注册
     *
     * @param uid   appid
     * @param name 渠道
     */
    public static void setAccount(String uid ,String name) {
        TDGAAccount account = TDGAAccount.setAccount(uid);
        account.setAccountName(name);
    }

    /**
     * 支付请求
     *
     * @param orderId               orderId
     * @param iapId                 iapId
     * @param currencyAmount        currencyAmount
     * @param currencyType          currencyType
     * @param virtualCurrencyAmount virtualCurrencyAmount
     * @param paymentType           paymentType
     */
    public static void onChargeRequest(String orderId, String iapId, String currencyAmount, String currencyType, String virtualCurrencyAmount, String paymentType) {
        TDGAVirtualCurrency.onChargeRequest(orderId, iapId, Double.valueOf(currencyAmount), currencyType, Double.valueOf(virtualCurrencyAmount), paymentType);
    }

    /**
     * 支付成功请求
     *
     * @param orderId orderId
     */
    public static void onChargeSuccess(String orderId) {
        TDGAVirtualCurrency.onChargeSuccess(orderId);
    }
}

File: frameworks/runtime-src/proj.ios_mac/ios/SDK/TalkingData/TalkingDataUtil.m
//  Created by lqn on 2017/8/23.
#import "TalkingDataUtil.h"

#import "TalkingDataGA.h"

@implementation ADD_CLASS(TalkingDataUtil)

+ (void) ADD_FUN(setAccount): (NSString *)uid name:(NSString *)name {
    TDGAAccount *tdgaAccount = [TDGAAccount setAccount:uid];
    [tdgaAccount setAccountName:name];
}

+ (void) ADD_FUN(onChargeRequest): (NSString *)orderId iapId:(NSString *)iapId currencyAmount:(NSString *)currencyAmount currencyType:(NSString *)currencyType virtualCurrencyAmount:(NSString *)virtualCurrencyAmount paymentType:(NSString *)paymentType {
    [TDGAVirtualCurrency onChargeRequst:orderId
                                  iapId:iapId
                         currencyAmount:[currencyAmount doubleValue]
                           currencyType:currencyType
                  virtualCurrencyAmount:[virtualCurrencyAmount doubleValue]
                            paymentType:paymentType];
}

+ (void) ADD_FUN(onChargeSuccess): (NSString *)orderId {
    [TDGAVirtualCurrency onChargeSuccess:orderId];
}

@end
